libcst/_nodes/tests/test_funcdef.py

```diff
@@ -365,6 +431,54 @@ class FunctionDefCreationTest(CSTNodeTest):
                 ),
                 "code": "def foo(*params: str, **kwparams: int): pass\n",
             },
+            # Test positional only params and positional params
+            {
+                "node": cst.FunctionDef(
+                    cst.Name("foo"),
+                    cst.Parameters(
+                        posonly_params=(cst.Param(cst.Name("bar")),),
+                        params=(cst.Param(cst.Name("baz")),),
+                    ),
+                    cst.SimpleStatementSuite((cst.Pass(),)),
+                ),
+                "code": "def foo(bar, /, baz): pass\n",
+            },
+            # Test positional only params and star args
+            {
+                "node": cst.FunctionDef(
+                    cst.Name("foo"),
+                    cst.Parameters(
+                        posonly_params=(cst.Param(cst.Name("bar")),),
+                        star_arg=cst.Param(cst.Name("baz")),
+                    ),
+                    cst.SimpleStatementSuite((cst.Pass(),)),
+                ),
+                "code": "def foo(bar, /, *baz): pass\n",
+            },
+            # Test positional only params and kwonly params
+            {
+                "node": cst.FunctionDef(
+                    cst.Name("foo"),
+                    cst.Parameters(
+                        posonly_params=(cst.Param(cst.Name("bar")),),
+                        kwonly_params=(cst.Param(cst.Name("baz")),),
+                    ),
+                    cst.SimpleStatementSuite((cst.Pass(),)),
+                ),
+                "code": "def foo(bar, /, *, baz): pass\n",
+            },
+            # Test positional only params and star kwargs
+            {
+                "node": cst.FunctionDef(
+                    cst.Name("foo"),
+                    cst.Parameters(
+                        posonly_params=(cst.Param(cst.Name("bar")),),
+                        star_kwarg=cst.Param(cst.Name("baz")),
+                    ),
+                    cst.SimpleStatementSuite((cst.Pass(),)),
+                ),
+                "code": "def foo(bar, /, **baz): pass\n",
+            },
```

Comment:
Can you add more advanced test case like `def name(p1, p2=None, /, p_or_kw=None, *, kw):`?

Reply:
We don't need to, we have plenty of tests that cover a host of permutations, we don't need to keep doubling the test count every time we add a new feature.